packages/ui/src/components/Dialog/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button, Heading } from '@utrecht/component-library-react';
+import type { ButtonProps } from '@utrecht/component-library-react';
+import { UtrechtIconClose } from '@utrecht/web-component-library-react';
+import classnames from 'classnames/bind';
+import { forwardRef } from 'react';
+import type { DetailedHTMLProps, DialogHTMLAttributes, ForwardedRef, PropsWithChildren, ReactNode } from 'react';
+import styles from './index.module.scss';
+
+export type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;
+export interface DialogProps extends DetailedHTMLProps<DialogHTMLAttributes<HTMLDialogElement>, HTMLDialogElement> {
+  footer?: ReactNode;
+  header?: {
+    title?: string;
+    headingLevel?: HeadingLevel;
```

Comment:
Misschien kunnen we die `HeadingLevel` type uit `HTMLHeading` halen? `HTMLHeadingProps['level']`?